src/index.js

```diff
@@ -6,6 +6,29 @@ export default function register(Component, tagName, propNames, options) {
 		inst._vdomComponent = Component;
 		inst._root =
 			options && options.shadow ? inst.attachShadow({ mode: 'open' }) : inst;
+
+		if (options && options.shadow && options.injectGlobalStyles) {
+			const defaults = {
+				target: document.head,
+				selector:
+					'style, link[rel="stylesheet"], link[rel="preload"][as="style"]',
+				filter: undefined,
+				observeOptions: { childList: true, subtree: true },
+			};
+
+			this.styleObserver = beginInjectingGlobalStyles(
```

Comment:
Super tiny nit: The prefix `begin*` somewhat implies that there is an `end*` function somewhere. Seeing that this function is only used once we can inline it and save about `14B`.